src/connection.c

```diff
@@ -700,8 +700,11 @@ int createNopollConnection(noPollCtx *ctx, server_list_t *server_list)
 	while (!g_shutdown)
 	{
 	  set_current_server (&conn_ctx);
-	  if (keep_trying_to_connect (&conn_ctx, &backoff_timer))
+	  if (keep_trying_to_connect (&conn_ctx, &backoff_timer)) {
+		// Don't reuse the redirect server during reconnect
+		free_server (&conn_ctx.server_list->redirect);
 		break;
+	  }
```

Comment:
OK, this will fix it, assuming we don't need the value of the redirect server once we have connected.

Reply:
@bill1600 is there a better way to reset the redirect url in all cases after v4 and v6 retry are done. As we need redirect url to reset after its tried one time.